controller/api/serializers.py

```diff
@@ -69,71 +70,60 @@ class AdminUserSerializer(serializers.ModelSerializer):
 
     class Meta:
         model = User
-        fields = ('username', 'is_superuser')
-        read_only_fields = ('username',)
+        fields = ['username', 'is_superuser']
+        read_only_fields = ['username']
 
 
 class AppSerializer(serializers.ModelSerializer):
     """Serialize a :class:`~api.models.App` model."""
 
-    owner = serializers.Field(source='owner.username')
-    id = serializers.SlugField(default=utils.generate_app_name)
-    url = serializers.Field(source='url')
-    structure = JSONFieldSerializer(source='structure', required=False)
+    owner = serializers.ReadOnlyField(source='owner.username')
+    structure = JSONFieldSerializer(required=False)
     created = serializers.DateTimeField(format=settings.DEIS_DATETIME_FORMAT, read_only=True)
     updated = serializers.DateTimeField(format=settings.DEIS_DATETIME_FORMAT, read_only=True)
 
     class Meta:
         """Metadata options for a :class:`AppSerializer`."""
         model = models.App
-
-    def validate_id(self, attrs, source):
```

Comment:
I'm happy to move this validation logic into the model, however, I want to make sure we're still throwing 400's with proper error messages if the validations fail.